src/queries/a11yValue.ts

```diff
@@ -0,0 +1,74 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { makeQueries } from './makeQueries';
+import type {
+  FindAllByQuery,
+  FindByQuery,
+  GetAllByQuery,
+  GetByQuery,
+  QueryAllByQuery,
+  QueryByQuery,
+} from './makeQueries';
+
+type A11yValue = {
+  min?: number;
+  max?: number;
+  now?: number;
+  text?: string;
+};
+
+export function matchObject<T extends Record<string, unknown>>(
+  prop: T | undefined,
+  matcher: T
+): boolean {
+  return prop
+    ? Object.keys(matcher).length !== 0 &&
+        Object.keys(prop).length !== 0 &&
+        !Object.keys(matcher).some((key) => prop[key] !== matcher[key])
```

Comment:
Using de Morgans law it seems easier to read. Is there any reason for favoring negated `some` over `every`?

```suggestion
        Object.keys(matcher).every((key) => prop[key] === matcher[key])
```

Reply:
BTW our docs could be updated with info that we require all params to matched in order for element to be matched.

Reply:
Agreed, your version is easier to read. I must admit that for the matchers, I purely copy pasted the existing code without trying to improve it

As for the docs, I can do it in a follow up PR, what do you think?